--- gnupg-1.3.1.orig/gnupg.c	2007-05-04 15:23:41.000000000 +0200
+++ gnupg-1.3.1/gnupg.c	2009-12-04 15:46:22.668179511 +0100
@@ -136,6 +136,7 @@
 static void gnupg_res_init(gnupg_object *intern TSRMLS_DC){
 	/* init the gpgme-lib and set the default values */
 	gpgme_ctx_t	ctx;
+	gpgme_check_version                             (0);
 	gpgme_new					(&ctx);
 	gpgme_set_armor				(ctx,1);
 	intern->ctx				=	ctx;
